src/Database/PostgreSQL/PQTypes/Checks.hs

```diff
@@ -160,6 +176,28 @@ checkUnknownTables tables = do
       ]
     else return mempty
 
+-- | Check that there's a 1-to-1 correspondence between the list of
+-- 'Table's and what's actually in the table 'table_versions'.
+checkExistanceOfVersionsForTables :: (MonadDB m, MonadLog m) => [Table] -> m ValidationResult
+checkExistanceOfVersionsForTables tables = do
+  runQuery_ $ sqlSelect "table_versions" $ do
+    sqlResult "name::text"
+  (existingTableNames :: [Text]) <- fetchMany runIdentity
+
+  let tableNames = map (unRawSQL . tblName) tables
+      absent     = existingTableNames L.\\ tableNames
+      notPresent = tableNames   L.\\ existingTableNames
+
+  if (not . null $ absent) || (not . null $ notPresent)
+    then do
+    mapM_ (logInfo_ . (<+>) "Unknown entry in 'table_versions':") absent
+    mapM_ (logInfo_ . (<+>) "Table not present in the 'table_versions':") notPresent
+    return . ValidationResult $
+      [ "Unknown entry in table_versions':" <+> T.intercalate ", " absent
```

Comment:
This will show both error messages even if one of the lists is empty, I'd write this e.g. as

```
   [ "Unknown entry in 'table_versions':" <+> absent' | not . null $ absent, let absent' = T.intercalate ", " absent ]
++ [ "Tables not present in the 'table_versions':" <+> notPresent' | not . null $ notPresent, let notPresent' = T.intercalate ", " notPresent ]
```